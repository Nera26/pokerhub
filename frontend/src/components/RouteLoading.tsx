import clsx from 'clsx';
import type { PropsWithChildren } from 'react';

interface RouteLoadingProps {
  className?: string;
}

export default function RouteLoading({
  children,
  className,
}: PropsWithChildren<RouteLoadingProps>) {
  return (
    <>
      <div className="h-20 bg-card-bg animate-pulse" />
      <main
        aria-busy="true"
        className={clsx(
          'container mx-auto px-4 sm:px-6 lg:px-8 pt-6 md:pt-8 pb-[calc(env(safe-area-inset-bottom)+72px)]',
          className,
        )}
      >
        {children}
      </main>
      <div className="h-20 bg-card-bg animate-pulse" />
    </>
  );
}
