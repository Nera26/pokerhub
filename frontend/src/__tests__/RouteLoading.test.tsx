import { render } from '@testing-library/react';
import RouteLoading from '@/components/RouteLoading';

describe('RouteLoading', () => {
  it('renders wrapper and children', () => {
    const { container } = render(
      <RouteLoading>
        <div className="placeholder" />
      </RouteLoading>,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders wrapper without children', () => {
    const { container } = render(<RouteLoading />);
    expect(container).toMatchSnapshot();
  });
});
