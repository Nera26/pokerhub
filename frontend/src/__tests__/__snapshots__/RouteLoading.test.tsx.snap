// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteLoading renders wrapper and children 1`] = `
<div>
  <div
    class="h-20 bg-card-bg animate-pulse"
  />
  <main
    aria-busy="true"
    class="container mx-auto px-4 sm:px-6 lg:px-8 pt-6 md:pt-8 pb-[calc(env(safe-area-inset-bottom)+72px)]"
  >
    <div
      class="placeholder"
    />
  </main>
  <div
    class="h-20 bg-card-bg animate-pulse"
  />
</div>
`;

exports[`RouteLoading renders wrapper without children 1`] = `
<div>
  <div
    class="h-20 bg-card-bg animate-pulse"
  />
  <main
    aria-busy="true"
    class="container mx-auto px-4 sm:px-6 lg:px-8 pt-6 md:pt-8 pb-[calc(env(safe-area-inset-bottom)+72px)]"
  />
  <div
    class="h-20 bg-card-bg animate-pulse"
  />
</div>
`;
